schoolopy/main.py

```diff
@@ -31,12 +31,12 @@ def _get(self, path):
         :param path: Path (following API root) to endpoint.
         :return: JSON response.
         """
+        response = self.schoology_auth.oauth.get(url='%s%s?limit=%s' % (self.api_host, path, self.limit), headers=self.schoology_auth._request_header(), auth=self.schoology_auth.oauth.auth)
+        response.raise_for_status()
         try:
-            response = self.schoology_auth.oauth.get(url='%s%s?limit=%s' % (self.api_host, path, self.limit), headers=self.schoology_auth._request_header(), auth=self.schoology_auth.oauth.auth)
-            response.raise_for_status()
             return response.json()
         except JSONDecodeError:
-            return {}
+            raise NoDataError(f'Get request to {response.url} failed with response {response.text}')
```

Comment:
I would recommend making the `response.text` a little more separated from the rest of the error message. For example:
```py
raise NoDataError(f'Get request to {response.url} failed: {response.text}')
```